docs/API/metadata.md

```diff
@@ -0,0 +1,43 @@
+## Dataset metadata
+
+A dataset is referenced by a formal resource ID, and also has a resource name for convenience. The Pbench server also maintains metadata about each dataset, which can help with searching and analysis. Some metadata is modifyable by an authenticated user, while other metadata is maintained internally by the server and can't be changed. Authenticated users can also add any additional metadata that might be of use.
+
+>__NOTE__: right now the ability to search and filter using metadata is limited, but our intent is to be able to use any defined metadata value both for searches and to filter the results of [datasets/list](V1/list.md).
+
+When a dataset is first processed, the Pbench server will populate basic metadata, including the creation timestamp, the owner of the dataset (the user authorization token used to authenticate the `PUT` from the agent via `pbench-results-move`), and the full contents of the dataset's `metadata.log` file inside the dataset tarball. These are all accessible under the `dataset` metadata key namespace.
+
+The Pbench server will also calculate a default deletion date for the dataset based on the owner's retention policy and the server administrator's retention policy along with some other internal management context. These are all accessible under the `server` metadata key namespace.
+
+Clients can also set arbitrary metadata through the "dashboard" and "user" metadata namespaces:
+* The "dashboard" namespace can only be modified by the owner of the dataset, and is visible to anyone with read access to the dataset.
+* The "user" namespace is private to each authenticated user, and even if you don't own a dataset you can set your own private "user" metadata to help you categorize that dataset and to find it again.
+
+Metadata namespaces are hierarchical, and are exposed as nested JSON objects. You can address an entire namespace, e.g., `dashboard` or `dataset` and retrieve the entire JSON object, or you can address nested objects or values using a dotted metadata key path like `dashboard.contact.email` or `dataset.metalog.pbench.script`.
```

Comment:
Is it worth noting what `dataset.metalog.pbench` would return?  A JSON object of all the sub-fields?

In a user JSON context, it would be the dictionary value.  But in other schemes, like what OpenShift offers for labels and annotations, you can have a value for `dataset.metalog.pbench` separate from the sub-objects.  We might want to document that we don't support that.